codec-http/src/main/java/io/netty/handler/codec/http/HttpVersion.java

```diff
@@ -45,6 +48,41 @@
     public static final HttpVersion HTTP_1_1 = new HttpVersion("HTTP", 1, 1, true, true);
 
     /**
+     * Internal method which allows to create known {@link HttpVersion} without create a temporary {@link String}
+     * to reduce object creation and so GC-pressure.
+     */
+    static HttpVersion valueOf(char[] chars, int startIndex, int endIndex) {
+        startIndex = HttpDecoderUtil.startIndex(chars, startIndex, endIndex);
+        endIndex = HttpDecoderUtil.endIndex(chars, startIndex, endIndex);
+        int size = endIndex - startIndex;
+        if (size == 0) {
```

Comment:
nit: I this I see why you didn't for this case...but consider using `switch(size)` to avoid multiple checks on the same integer variable.